src/main/java/io/jenkins/plugins/pipelinegraphview/consoleview/PipelineConsoleViewAction.java

```diff
@@ -312,4 +313,147 @@ public RunDetailsCard getRunDetailsCard() {
     public boolean isShowGraphOnBuildPage() {
         return PipelineGraphViewConfiguration.get().isShowGraphOnBuildPage();
     }
+
+    public boolean isBuildable() {
+        return run.getParent().isBuildable();
+    }
+
+    public boolean isBuildInProgress() {
+        return run.isBuilding();
+    }
+
+    public Permission getPermission() {
+        return Item.BUILD;
+    }
+
+    public Permission getCancelPermission() {
+        return Item.CANCEL;
+    }
+
+    public Permission getConfigurePermission() {
+        return Item.CONFIGURE;
+    }
+
+    public String getBuildDisplayName() {
+        return run.getDisplayName();
+    }
+
+    public String getBuildFullDisplayName() {
+        return run.getFullDisplayName();
+    }
+
+    public boolean isRebuildAvailable() {
+        Plugin rebuildPlugin = Jenkins.get().getPlugin("rebuild");
+        if (rebuildPlugin != null && rebuildPlugin.getWrapper().isEnabled()) {
+            // limit rebuild to parameterized jobs otherwise it duplicates rerun's behaviour
+            return run.getParent().getProperty(ParametersDefinitionProperty.class) != null;
+        }
+        return false;
+    }
+
+    public boolean isRestartFromStageAvailable() {
+        RestartDeclarativePipelineAction action = run.getAction(RestartDeclarativePipelineAction.class);
+        if (action != null) {
+            return action.isRestartEnabled();
+        }
+        return false;
+    }
+
+    /**
+     * Handles the rerun request using ReplayAction feature
+     */
+    @RequirePOST
+    @JavaScriptMethod
+    public boolean doRerun() throws IOException, ExecutionException {
+        if (run != null) {
+            run.checkAnyPermission(Item.BUILD);
```

Comment:
while you are here, is there a reason this isn't this? (might be why codeql is complaining)
```suggestion
            run.checkPermission(Item.BUILD);
```